### PR TITLE
US931 data model publication statement

### PR DESCRIPTION
**Pull request by rerowep**
data model: implement publication statement for ebooks
    * Implements the publication statement transformation for ebooks.
    * Uniforms the function names to snake_case (PEP8).
    * Refactors Marc21 punctation handling.
    * Fixes copyright date transformation.

    ui: correct document brief views
    * Adds publication statement to admin and public brief document view.

    ui: adapt editor according to publication statement data model
    * Removes publication date field

    ES: adapt ES mapping for publication statement
    * Adapts the ES mapping for publication statement.

    data model: implement publication statement transformation for BNF
    * Implements the publication statement transformation for BNF.
    * Adds base class for marc dojson transformation.
    * Adds colored message functions to `setup` and `bootstrup` scripts

    UI:display the publication statement
    * Displays the publication statement on detailed view.

    template: correct email templates
    * Removes item at desk templates for patron and according methods and tests.

    data model: implement publication statement transformation
    * Implements the publication statement transformation.
    * Adds the dojson overdo class with blob.
    * Adds functions to manipulate the Marc21 blob.
    * Adapts the fixtures.
    * Adds parameters to the setup for generating items and holdings data.

    data: correct language in big document file
    * Corrects notice 704 tag 008 esp->sp.

### CODE REVIEW

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 1]
```diff
@@ -0,0 +1,925 @@
...
+            "title": "Qualifier",
+            "description": "Information about the person, ie her profession. Helpful to disambiguate people.",
+            "type": "string"
+          }
+        }
+      }
+    },
+    "copyrightDate": {
+      "title": "Copyright Date",
+      "description": "Copyright Date of the resource.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 2]
```diff
@@ -0,0 +1,925 @@
...
+      "minItems": 1,
+      "items": {
+        "type": "string",
+        "pattern": "^([\u00a9\u2117]\\s)?\\d{4}.*",
+        "minLength": 4
+      }
+    },
+    "provisionActivity": {
+      "title": "Provision Activity",
+      "description": "Provision Activity of the resource.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 3]
```diff
@@ -0,0 +1,925 @@
...
+            "enum": [
+              "bf:Publication",
+              "bf:Manufacture",
+              "bf:Distribution",
+              "bf:Production"
+            ]
+          },
+          "statement": {
+            "title": "Statement of od place and/or agent of publication",
+            "description": "Statement of place and/or agent of the resource.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 4]
```diff
@@ -0,0 +1,925 @@
...
+                  "title": "Type",
+                  "type": "string",
+                  "enum": [
+                    "bf:Place",
+                    "bf:Agent"
+                  ]
+                },
+                "canton": {
+                  "title": "Canton",
+                  "description": "Canton.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 5]
```diff
@@ -0,0 +1,925 @@
...
+                  "type": "array",
+                  "minItems": 1,
+                  "items": {
+                    "type": "string",
+                    "$ref": "#/definitions/canton"
+                  }
+                },
+                "country": {
+                  "title": "Country",
+                  "description": "Country.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 6]
```diff
@@ -0,0 +1,925 @@
...
+                  "type": "string",
+                  "$ref": "#/definitions/country"
+                },
+                "label": {
+                  "title": "Label",
+                  "description": "Provision Activity place and/or agent.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Is this `description` actually the one we want? Should it be removed?

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 7]
```diff
@@ -0,0 +1,925 @@
...
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "value": {
+                        "title": "Place and/or agent name",
+                        "description": "Place and/or agent name.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 8]
```diff
@@ -0,0 +1,925 @@
...
+                        "type": "string",
+                        "minLength": 2
+                      },
+                      "language": {
+                        "title": "Language",
+                        "description": "Language.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added.

**Comment by NicolasLabat** [comment]
In the detailed view the heading for the publication statement is "bf:Publication:" --> to correct in "Publication:"

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 9]
```diff
@@ -0,0 +1,925 @@
...
+      "type": "array",
+      "minItems": 1,
+      "items": {
+        "type": "string",
+        "minLength": 3
+      }
+    },
+    "identifiedBy": {
+      "title": "Identifiers",
+      "description": "Persistent identifiers of the resource, ie reroID and ISBN.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: This isn't accurate anymore. Remove the `description` field.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json [hunk 10]
```diff
@@ -0,0 +1,925 @@
...
+        "type": "object",
+        "required": [
+          "type",
+          "value"
+        ],
+        "additionalProperties": false,
+        "properties": {
+          "type": {
+            "title": "Type",
+            "description": "Type of the value.",
```
Comment (iGormilhit) [inline rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Remove this description, less one have a more informative sentence to be added. Or something like `Identifiers type.` But I think it's better to remove it.

File: rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json (hunk 6 above)
Comment (rerowep) [reply rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Don't know

Comment (rerowep) [reply rero_ils/modules/documents/jsonschemas/documents/document-v0.0.1_src.json]: Ok I delete it!

**Comment by NicolasLabat** [comment]
Same for "bf:Manufacture"

**Comment by pronguen** [comment]
> 
> 
> In the detailed view the heading for the publication statement is "bf:Publication:" --> to correct without "bf:". Is the heading "Publication" correct ?

Is addressed by the translations

**Comment by NicolasLabat** [comment]
In case of a 264 like $a : $b ; $a : $b the ; is replaced by : but has be to kept.
Ex :
264 _1 $a [Lieu de publication non identifié] : $b SZ ; $a [Paris] : $b Arte France [prod.], $c [2001] 
is now displayed : 
[Lieu de publication non identifié] : SZ : [Paris] : Arte France [prod.], [2001]
Must be :
[Lieu de publication non identifié] : SZ ; [Paris] : Arte France [prod.], [2001]

**Pull request merged**